### ISSUE TITLE
Duplicate class error on Android for ffmpeg

### ISSUE THREAD

**Issue by ISnowFoxI** [maitrungduc1410/react-native-video-trim#48]
First of all, thank you for this library.

I was using `ffmpeg-kit-react-native` to edit video metadata on Android. I also needed a way to trim videos, so I used this library. I would get multiple `Duplicate class` errors when building for Android because of mismatched FFmpeg packages in the project. Using

```
ext {
    ffmpegKitPackage = "<package name>"
}
```
didn't help me.

I suggest adding additional steps to the readme for conflicting install cases. My solution was editing the `android/app/build.gradle` file like so:

```
dependencies {
    ...
     modules {
        module('com.arthenica:ffmpeg-kit-https') {
            replacedBy 'com.arthenica:ffmpeg-kit-min'
        }
    }
}
```

I found the above solution in this blog post:
https://tomgregory.com/gradle/how-to-exclude-gradle-dependencies/#option-3-module-replacement-exclude-alternative

I've also opened an issue in `ffmpeg-kit` repo here -> https://github.com/maitrungduc1410/react-native-video-trim/issues/new for further discussion if needed.

**Comment by maitrungduc1410** [comment]
hi, `ffmpeg-kit-react-native` also uses ffmpegKit under the hood, so we can just sync the version between both and it should be fine (technically)

```
buildscript {
    ext {
        ffmpegKitPackage = "full" // default "min"

        ffmpegKitPackageVersion = "5.1.LTS" // default 6.0-2
    }
```
You just need to find out which ffmpegKit version of `ffmpeg-kit-react-native`, then use the above code

**Comment by maitrungduc1410** [comment]
upon checking latest source of [ffmpeg-kit-react-native](https://github.com/arthenica/ffmpeg-kit/blob/main/react-native/android/build.gradle), the default version is `com.arthenica:ffmpeg-kit-https:6.0.2`

then to make it work with my lib you should edit `android/build.gradle` like:

```
buildscript {
    ext {
        ffmpegKitPackage = "https"

        ffmpegKitPackageVersion = "6.0.2"
    }
```

**Comment by ISnowFoxI** [comment]
@maitrungduc1410  thank you for getting back, I think I tested this and it didn't work. That said, I'll test it once more and get back to you.  I was trying to use the `min` package but It has been over a month since so I can't remember the exact details.

**Comment by ISnowFoxI** [comment]
I am going to close this issue as I haven't had the time to test it yet, I'll re-open it if I have the time to test it. Thank you

**Comment by vjkrishna21** [comment]
> upon checking latest source of [ffmpeg-kit-react-native](https://github.com/arthenica/ffmpeg-kit/blob/main/react-native/android/build.gradle), the default version is `com.arthenica:ffmpeg-kit-https:6.0.2`
> 
> then to make it work with my lib you should edit `android/build.gradle` like:
> 
> ```
> buildscript {
>     ext {
>         ffmpegKitPackage = "https"
> 
>         ffmpegKitPackageVersion = "6.0.2"
>     }
> ```

I tried the steps you've provided, and I got a different issue. However @ISnowFoxI 's solution worked for me. Adding the error below


<img width="1155" alt="Screenshot 2024-09-29 at 11 44 01 PM" src="https://github.com/user-attachments/assets/621ed2cb-c613-48c6-873a-74a0e2c36e7f">

**Comment by maitrungduc1410** [comment]
@vjkrishna21 so does it work now?

**Comment by vjkrishna21** [comment]
Yes. The debug build worked. I'll make sure to keep you posted on the release build. Can you maybe try to create a new example app on yourside with react-native-video-trim and ffmpeg-kit-react-native as dependencies to understand the exact issue and maybe update the README with an appropriate fix?

**Comment by maitrungduc1410** [comment]
@techbyvj have you checked my [README](https://github.com/maitrungduc1410/react-native-video-trim?tab=readme-ov-file#naming-conflict-with-ffmpeg-kit-react-native)?

In there I wrote solution to solve this issue. And to make sure it's correct, I did generate new project just now and tried.

Initial `pod install` (after yarn add react-native-video-trim + ffmpeg-kit-react-native):

<img width="1368" alt="Screenshot 2024-10-01 at 5 30 55 PM" src="https://github.com/user-attachments/assets/d8c97587-3d13-46e0-ae44-27ec30b0d420">

With the guide in the README we can find out that ffmpeg-kit-react-native is using `https 6.0.2`

Then run this:
```
FFMPEGKIT_PACKAGE=https FFMPEG_KIT_PACKAGE_VERSION=6.0.2 pod install
```
And it's successful now:

<img width="1225" alt="Screenshot 2024-10-01 at 5 34 51 PM" src="https://github.com/user-attachments/assets/686775a6-aa1b-4f18-a585-944e5420f80a">
